README.md

```diff
@@ -1,4 +1,274 @@
-Direct Embedding for Scala [![Build Status](https://travis-ci.org/directembedding/directembedding.png?branch=master)](https://travis-ci.org/directembedding/directembedding)
+DirectEmbedding [![Build Status](https://travis-ci.org/directembedding/directembedding.png?branch=master)](https://travis-ci.org/directembedding/directembedding)
 ===============
 
-A library for macro based direct embedding of DSLs.
+An experimental macro-based library for Embedded DSLs.
+
+# DirectEmbedding for DSLs
+
+
+DirectEmbedding is an experimental project that attempts to provide direct embedding in Scala for Domain Specific Languages (DSLs). Although, this project is currently exploratory, this library could become a very useful tool. For instance, it could replace the complex shadow embedding of [Yin-Yang](https://github.com/scala-yinyang/scala-yinyang/tree/cbfaf02405c5e273498c15ce943965eeba8afa31) library in existing or future projects such as [Slick](http://slick.typesafe.com/).
+
+This library provides **one** effortless logic for the reification of embedded DSLs in Scala. This comes along many **advantages**:
+
+- no knowledge of the [Reflection API](http://en.wikipedia.org/wiki/Reflection_%28computer_programming%29)
+- no overloading resolution
+- no dependence with types and count of arguments
+- no verbosity
+- no duplication of code
+
+Our solution makes use of [annotations](http://docs.scala-lang.org/overviews/reflection/annotations-names-scopes.html) and [macros](http://www.scala-lang.org/api/2.11.0/scala-reflect/index.html#scala.reflect.macros.blackbox.Context) to achieve this goal.
+
+##Table of contents:
+
+1. [Overview of the concept](#overview)
+2. [Details](#details)
+3. [Project Structure](#project-structure)
+4. [Usage](#usage)
+5. [References](#references)
+6. [Licence](#licence)
+
+--
+
+# Overview
+Writing embedding DSLs implies to go though the difficult task of reification. 
+Reification means when a function *take(x: Int): Query[T]* is written in Scala it will be converted to a corresponding DSL object *Take*. *Take* is the Intermediate Representation (IR) of the function *take()*. The naive method would code conditional statements in order to **identify** which function has been called. Identifying is strenuous and redundant because it must check many criteria, take in account aspects such as overriding of the function and this for all functions or classes. It is not a best practices and it creates as much logics of reification than there is cases to reify.
```

Comment:
"Reification means when" ->
Reification is conversion of domain-specific and Scala operations to an intermediate representation (IR). For example, take...